docs/compiler-exceptions.rst

```diff
@@ -0,0 +1,276 @@
+             3 struct Bid:
+        ---> 4   blindedBid bytes32
+        ---------------------------^
+             5   deposit: wei_value
+
+    The terminal output of a syntax error will generally show exactly where it happened.  In this case there is a semi
+    colon missing after ``blindedBid`` in the declaration of the struct.
+
+.. py:exception:: TypeMismatchException
+
+.. code-block:: python3
+
+    @public
+    def foo():
+    a: uint256 = 42
+    b: bytes32 = a
+    vyper.exceptions.TypeMismatchException: line 4:4 Invalid type, expected: bytes32
+         3     a: uint256 = 1
+    ---> 4     b: bytes32 = a
+    -----------^
+
+This exception occurs when a variable is assigned a value that is inconsistent with the type.  ``bytes32`` is the declaration of an array but only contains the single value of a character.
+
+
+.. py:exception:: VariableDeclarationException
+
+    .. code-block:: python3
+
+        # Final auction state
+        highestBid: public(wei_value)
+        highestBidder: public(address)
+    
+        @private
+        def placeBid(bidder: address, value: wei_value) -> bool:
+        # If bid is less than highest bid, bid fails
+        if (value <= self.highstBid):
+            return False
+
+    ``VariableDeclarationException`` is a compiling error in which a variable is being used that has not been declared.
+
+    .. code-block:: bash
+
+        $ vyper blind_auction.vy
+        Error compiling: blind_auction.vy
+        /usr/lib/python3/dist-packages/apport/report.py:13: DeprecationWarning: the imp module is deprecated in favour of             importlib; see the module's documentation for alternative uses import fnmatch, glob, traceback, errno, sys, atexit,           locale, imp
+        vyper.exceptions.VariableDeclarationException: line 79:17 Persistent variable undeclared: highstBid
+             78     # If bid is less than highest bid, bid fails
+        ---> 79     if (value <= self.highstBid):
+        ------------------------^
+             80         return False
+
+    ``self.highestBid`` is using a misspelled modified version of the public variable ``highestBidder``.
+
+.. py:exception:: VersionException
+
+    .. code-block:: python3
+
+        @version 0.1.0b13
+
+    This exception will happen when a version pragma is being compiled with a future compiler.  Version pragma
+    declaration should be the first line of the file.
+
+
+.. py:exceptions:: ZeroDivisionException
+
+    This exception will occur when a divide by zero or ``modulo`` zero situation arises.  The terminal will point out the line 
+    which contains the error.
+
+CompilerPanic
+=============
+
+.. py:exception:: CompilerPanicException
+
+    .. code-block:: python3
+
+        $ vyper v.vy 
+        Error compiling: v.vy
+        vyper.exceptions.CompilerPanic: Number of times repeated must be a constant nonzero positive integer: 0 Please create an     issue.
```

Comment:
```suggestion
        vyper.exceptions.CompilerPanic: Number of times repeated must be a
        constant nonzero positive integer: 0 Please create an issue.
```

Because code blocks don't wrap, this is making me use a horizontal scrollbar on my gigantic 11 inch laptop screen :laughing:  I know it's not an exact representation of the error if we add a newline, but I think it's still accurate enough - and preferable to scrolling.

Reply:
Fixed